crates/sui-core/src/authority.rs

```diff
@@ -721,6 +721,40 @@ impl AuthorityState {
         Ok((inner_temp_store, signed_effects))
     }
 
+    pub async fn call_transaction(
```

Comment:
Could you check how similar this is to `prepare_certificate` and try to reduce code duplication?